File: dart/matching-brackets/lib/matching_brackets.dart
class MatchingBrackets {
  bool isPaired(String s) {
    var bracket_map = { '{': '}', '[': ']', '(': ')' };
    List<String> stack = new List<String>();
    for (var c in s.split('')) {
      if (bracket_map.containsKey(c)) {
        stack.add(bracket_map[c]);
      } else if (bracket_map.values.contains(c)) {
        if (stack.length == 0 || stack.removeLast() != c) {
          return false;
        }
      }
    }
    return stack.length == 0;
  }
}

File: kotlin/matching-brackets/src/main/kotlin/MatchingBrackets.kt
object MatchingBrackets {

    val brackets = mapOf('[' to ']', '{' to '}', '(' to ')' )

    fun isValid(input: String): Boolean {
        val stack = arrayListOf<Char>()
        for (c in input) {
            if (brackets.values.contains(c)) {
                if (stack.size == 0 || stack.removeAt(stack.lastIndex) != c) {
                    return false
                }
            } else if (brackets.keys.contains(c)) {
                stack.add(brackets[c]!!)
            }
        }
        return stack.size == 0
    }
}
